### PR TITLE
Fix cancelling drag behaviour when using the esc key

### PR DESCRIPTION
**Pull request by jacobscarter**
Fixes issue https://github.com/angular-ui-tree/angular-ui-tree/issues/799

### CODE REVIEW

**Comment by jacobscarter** [comment]
@StefanCodes @Voles this PR of mine is ready for review, should be an easy one

File: source/directives/uiTreeNode.js [hunk 1]
```diff
@@ -48,7 +48,7 @@
             scope.init(controllersArr);
 
             scope.collapsed = !!UiTreeHelper.getNodeAttribute(scope, 'collapsed') || treeConfig.defaultCollapsed;
-			scope.expandOnHover = !!UiTreeHelper.getNodeAttribute(scope, 'expandOnHover');
+			      scope.expandOnHover = !!UiTreeHelper.getNodeAttribute(scope, 'expandOnHover');
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: Extra indentation is probably added by your IDE?

File: source/directives/uiTreeNode.js [hunk 2]
```diff
@@ -68,7 +68,7 @@
               }
             });
 
-			scope.$watch('expandOnHover', function (val) {
+			      scope.$watch('expandOnHover', function (val) {
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: See above comment.

File: source/directives/uiTreeNode.js [hunk 3]
```diff
@@ -572,6 +572,12 @@
                 }
               };
             })();
+            
+            keydownHandler = function (e) {
+              if (e.keyCode == 27) {
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: I know the original code also contained `==`, but please use the triple equal sign as much as possible.

File: source/directives/uiTreeNode.js [hunk 4]
```diff
@@ -572,6 +572,12 @@
...
+                dragEndEvent(e);
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: Perfect. So we shouldn't have called the `dragEnd` method directly, but instead call `dragEndEvent` first?

File: source/directives/uiTreeNode.js [hunk 5]
```diff
@@ -598,6 +604,7 @@
               angular.element($document).bind('mouseup', dragEndEvent);
               angular.element($document).bind('mousemove', dragMoveEvent);
               angular.element($document).bind('mouseleave', dragCancelEvent);
+              angular.element($document).bind('keydown', keydownHandler);
```
Comment (Voles) [inline source/directives/uiTreeNode.js]: Perfect, this is now consistent with other bindings 👍

**Comment by Voles** [comment]
Nice fix! I've added some minor comments.

File: source/directives/uiTreeNode.js (hunk 4 above)
Comment (jacobscarter) [inline source/directives/uiTreeNode.js]: Correct, dragEnd doesn't have scope but dragEndEvent does

**Comment by jacobscarter** [comment]
@Voles update per your comments

**Pull request merged**